src/sqlite3gen.cpp

```diff
@@ -138,55 +206,71 @@ const char * schema_queries[][2] = {
+      "\tmemberdef_rowid        INTEGER NOT NULL REFERENCES memberdef, -- reimplementing memberdef id.\n"
+      "\treimplemented_rowid    INTEGER NOT NULL REFERENCES memberdef, -- reimplemented memberdef id.\n"
+      "\tUNIQUE(memberdef_rowid, reimplemented_rowid) ON CONFLICT IGNORE\n"
+    ");\n"
   },
   { "compounddef",
     "CREATE TABLE IF NOT EXISTS compounddef (\n"
-      "\t-- class/struct definitions.\n"
-      "\trowid        INTEGER PRIMARY KEY AUTOINCREMENT NOT NULL,\n"
-      "\tname         TEXT NOT NULL,\n"
-      "\tkind         TEXT NOT NULL,\n"
-      "\trefid        INTEGER NOT NULL,\n"
-      "\tprot         INTEGER NOT NULL,\n"
-      "\tid_file      INTEGER NOT NULL,\n"
-      "\tline         INTEGER NOT NULL,\n"
-      "\tcolumn       INTEGER NOT NULL\n"
-      ");"
-  },
-  { "basecompoundref",
-    "CREATE TABLE IF NOT EXISTS basecompoundref (\n"
-      "\trowid        INTEGER PRIMARY KEY AUTOINCREMENT NOT NULL,\n"
-      "\tbase         TEXT NOT NULL,\n"
-      "\tderived      TEXT NOT NULL,\n"
-      "\trefid        INTEGER NOT NULL,\n"
-      "\tprot         INTEGER NOT NULL,\n"
-      "\tvirt         INTEGER NOT NULL\n"
-      ");"
+      "\t-- Class/struct definitions.\n"
+      "\trowid                INTEGER PRIMARY KEY NOT NULL,\n"
+      "\tname                 TEXT NOT NULL,\n"
+      "\ttitle                TEXT,\n"
+      // probably won't be empty '' or unknown, but the source *could* return them...
+      "\tkind                 TEXT NOT NULL, -- 'category' 'class' 'constants' 'dir' 'enum' 'example' 'exception' 'file' 'group' 'interface' 'library' 'module' 'namespace' 'package' 'page' 'protocol' 'service' 'singleton' 'struct' 'type' 'union' 'unknown' ''\n"
+      "\tprot                 INTEGER,\n"
+      "\tfile_id              INTEGER NOT NULL REFERENCES path,\n"
+      "\tline                 INTEGER NOT NULL,\n"
+      "\tcolumn               INTEGER NOT NULL,\n"
+      "\theader_id            INTEGER REFERENCES path,\n"
+      "\tdetaileddescription  TEXT,\n"
+      "\tbriefdescription     TEXT,\n"
+      "\tFOREIGN KEY (rowid) REFERENCES refid (rowid)\n"
+    ");"
   },
-  { "derivedcompoundref",
-    "CREATE TABLE IF NOT EXISTS derivedcompoundref (\n"
-      "\trowid        INTEGER PRIMARY KEY AUTOINCREMENT NOT NULL,\n"
-      "\tbase         TEXT NOT NULL,\n"
-      "\tderived      TEXT NOT NULL,\n"
-      "\trefid        INTEGER NOT NULL,\n"
-      "\tprot         INTEGER NOT NULL,\n"
-      "\tvirt         INTEGER NOT NULL\n"
-      ");"
+  { "compoundref",
+    "CREATE TABLE IF NOT EXISTS compoundref (\n"
+      "\t-- Inheritance relation.\n"
+      "\trowid          INTEGER PRIMARY KEY AUTOINCREMENT NOT NULL,\n"
+      "\tbase_rowid     INTEGER NOT NULL REFERENCES compounddef,\n"
+      "\tderived_rowid  INTEGER NOT NULL REFERENCES compounddef,\n"
+      "\tprot           INTEGER NOT NULL,\n"
+      "\tvirt           INTEGER NOT NULL,\n"
+      "\tUNIQUE(base_rowid, derived_rowid)\n"
+    ");"
```

Comment:
`compoundref` folds in both inheritance relation tables (`basecompoundref` and `derivedcompoundref`) into single bi-directional association table.